webapp/src/marked.tsx

```diff
@@ -350,6 +352,105 @@ export class MarkedContent extends data.Component<MarkedContentProps, MarkedCont
             })
     }
 
+    // Renders collapsable hints starting with "~hint" and ending with "hint~".
+    // Example input:
+    // -------------------------------
+    // ~hint This is the hint summary.
+    //
+    //    This is the hint content
+    //
+    // hint~
+    // -------------------------------
+    //
+    // Example output:
+    // -------------------------------
+    // <details><summary>This is the hint summary</summary>
+    //   This is the hint content
+    // </details>
+    // -------------------------------
+    private renderAccordianHints(content: HTMLElement) {
+        const hintBeginRegex = /^\s*~hint\s*(.+)/i;
+        const hintEndRegex = /^\s*hint~.*/i;
+
+        // Processing each hint modifies the node order and count of 'content.childnodes'.
+        // Keep reprocessing content.childnodes until all hints have been processed.
+        let hintFound: boolean = false;
+
+        do {
+            hintFound = false;
+            // The hint summary element (i.e. <summary>text</summary>).
+            let hintSummary: HTMLElement = null;
+            // The element that signifies the beginning of the hint (i.e. <p>~hint (hint summary)</p>).
```

Comment:
would be nice to have this logic included for docs view (e.g. localhost:3232/tutorials/chase-the-pizza) -- that logic lives over in https://github.com/microsoft/pxt/blob/master/pxtrunner/renderer.ts#L1302 . Currently it shows up as plaintext: 
![image](https://user-images.githubusercontent.com/5615930/176067753-97919c94-c7c8-4f12-8e2d-de6680c2277b.png)

(but this is also probably fine to handle separately / as a follow up, that view often ends up a little neglected to be honest and it's not the primary scenario)

Reply:
I recommend handling this scenario in a separate PR.

Reply:
Yup, agreed